### ISSUE TITLE
Cast-When-Held Free Action failing vs ancient lich

### ISSUE THREAD

**Issue by JayH2971** [Interkarma/daggerfall-unity#2338]
**Describe the issue**

This save file comes from another user, which I tested without mods. The character's torc has a Cast When Held effect of Free Action, equipped at load time. However, the lich in front of the character uses a spell (Wizard Rend?) which can still paralyze this character.

**To Reproduce**

1. Approach the lich
2. If it uses a Wizard Rend spell, it should paralyze you, as it did me

**Expected behavior**

CWH Free Action should provide protection against paralysis spells.

[SAVE113.zip](https://github.com/Interkarma/daggerfall-unity/files/8704271/SAVE113.zip)

**Comment by jeniferirwin** [comment]
The 'Force Bolt' spell says that it can cause paralysis too. There do not appear to be any mobs in the base game that cast this, but mod authors might have mobs that cast it. Force Bolt should probably be looked at for this bug as well.

**Comment by JayH2971** [comment]
I'm noticing a few times with a similar object CWH, that it says "spell effect failed" when equipping or fast traveling. It may be that the effect isn't taking hold when that happens.

**Comment by Interkarma** [comment]
I've looked into this one. What's happening is the Cast When Held: Free Action enchantment is actually casting Cure Paralyzation instead. Most likely an improper casting of effect IDs in Cast When Held specifically. This doesn't apply to the spell-based Free Action effect, only the Cast When Held version.

I'll investigate further and resolve. Thanks for raising.

**Comment by Interkarma** [comment]
This problem is the result of (probably to the surprise of nobody) classic game data. Here's what's happening:

Cast When Held uses a lookup table to reference classic spell type/subType from SPELLS.STD. This entry has the correct spell ID, but when enumerating the effects read from classic data the effect referenced is type=3, subType=2 - which is Cure Paralyzation not Free Action effect. Rather, the Free Action effect is correctly type=26, subType=-1.

Simply put, the spell record for Free Action in classic data is referencing the wrong effect. The reason this doesn't happen with the spell version is that's all happening in DFU with its native key values and no read from classic spell data is required. The disconnect is casting bad data from classic into DFU.

The best way to fix this is probably the most direct - patch bad data on the fly. I'll have a think about it and implement fix soon.
